sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosResponse.java

```diff
@@ -15,9 +15,9 @@
  */
 public class CosmosResponse<T extends Resource> {
     private T properties;
-    protected final ResourceResponse resourceResponseWrapper;
+    protected final ResourceResponse<?> resourceResponseWrapper;
 
-    protected CosmosResponse(ResourceResponse resourceResponse) {
+    protected CosmosResponse(ResourceResponse<?> resourceResponse) {
```

Comment:
API change for @kushagraThapar and @moderakh to review.

Reply:
This looks good to me.